### DISCUSSION TITLE
[v5] Missing label IDs in _locales?

### DISCUSSION THREAD

**Discussion by Qjo1** [mbnuqw/sidebery#726] [Q&A] [answered]
I created a full translation for this great extension, however in the sidebar's right click menus there are terms that aren't declared in _locales. 

As far as I could tell it concerns the "move to" and "reopen in" sub-menus for tabs and "open in" and "sort" sub-menus for bookmarks.  

Example: in the installed (English) extension it looks like this: 

> Move to  > New Window

But in the locales (dict.common) there is only: 

```
  'menu.tab.move_to_sub_menu_name': {
    en: 'Move to',},

  'menu.tab.move_to_new_window': {
    en: 'Move to new window',},
```
Which should (but doesn't) make the menu look like this:

> Move to  > Move to new window

I realized this is because the definitions in dict.common are for those menus' description in the settings but not in the actual menu. Well, for English that is. If I provide a translation there then it is used for the menus as well, which tells me the menus are defined elsewhere and the settings descriptions are only a fall-back.  But where are they defined? 

Edit: 
Okay so I scanned through all files but I found nothing. 
So...I could of course "fix" the definitions in dict.common to get the menus right, but I don't think this is the intention. @mbnuqw Could you clarify how I should approach this? The translation is complete, otherwise. 

Cheers

**Comment by mbnuqw** [comment, marked as answer]
Sorry for the late response and thanks again for the PR!

The label of sub-item in context menu will be automatically trimmed (and capitalized) if it starts with the parent item's label. This behavior was added because user can edit context menu and place items in the top level list as well as in the sub-list with the custom label of parent item.  

So each menu item's label should contain full name, e.g:
#### Case 1 - default
- `Move to:`
  - `Move to new window` - will be trimmed to `New window`
#### Case 2 - user moved `Move to new window` to the top level list
- `Move to new window` - will not be trimmed and print `Move to new window`
#### Case 3 - user changed parent item's title
- `Moving...`
  - `Move to new window` - will not be trimmed because it doesn't start with `Moving...`

Does it possible to write something like `Verschieben in neues Fenster` or maybe `Verschieben in: Neues Fenster` instead of `In neues Fenster verschieben`?

**Reply by Qjo1** [reply]
No problem, I'm happy to contribute this way since I can't code much. Also, I just figured I can do the PR now and change those parts later if there's a cleaner way, that's why I sent it.

So thanks for your detailed explanation, I understand now (and it makes me think I could've figured this out myself but oh well). Something like you suggested should be fine. I'll have a look at it again and update accordingly.

Btw: I could also provide translations for the AMO files if you're interested.

**Reply by mbnuqw** [reply]
Thank you. It is only my fault that I didn't provide neither documentation nor comments in code describing how translation works in sidebery.

> Btw: I could also provide translations for the AMO files if you're interested.

It would be nice but AMO description and summary text for v5 are not yet ready (I'll add them to the repo later).

**Reply by Qjo1** [reply]
Alright, I'll tackle them once you've added the new ones then.
